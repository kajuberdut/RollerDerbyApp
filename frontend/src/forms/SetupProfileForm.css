.ProfileForm {
    justify-content: center;
    margin: auto;
    margin-top: 6em;
    width: 50em;
}

