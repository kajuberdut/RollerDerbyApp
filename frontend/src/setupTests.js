// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import './setupTests.js';

// https://stackoverflow.com/questions/32911630/how-do-i-deal-with-localstorage-in-jest-tests
const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    clear: jest.fn()
  };
  global.localStorage = localStorageMock;