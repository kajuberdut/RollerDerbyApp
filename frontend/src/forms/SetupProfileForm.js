import React, { useContext, useState } from "react";
import FastApi from "../Api";
// import { useHistory } from "react-router-dom/cjs/react-router-dom.min";
// import UserContext from "../../repeated/UserContext";
import { useNavigate } from "react-router-dom";
import "./SetupProfileForm.css"
import {
    Card,
    CardBody,
    CardTitle,
    Form,
    FormGroup,
    Label, 
    Input,
    Button,
  } from "reactstrap";
  import PropTypes from 'prop-types';

/** 
 * Form for creating a user or updating a logged in user.
 */

const SetupProfileForm = ({signup, update}) => {

  /** Set user, history and initial state and set valid, invalid, and error message in state */

  // const { user, setUser } = useContext(UserContext);
//   const history = useHistory();
//   const [ valid, setValid ] = useState(false);
//   const [ invalid, setInvalid ] = useState(false);
//   const [errorMessage, setErrorMessage] = useState([]);
const [dropdownOpen, setDropdownOpen] = useState(false);
// const [file, setFile] = useState<File | undefined>();
// const [preview, setPreview] = useState<string | undefined>();
const navigate = useNavigate();

let user = "SockHer Blue"  

  /** 
   * Sets Initial State of Form
  */



let INITIAL_STATE = { derbyName: user.derbyName, firstName: "", lastName: "", email: "",  facebookName: "", about: "", primNum: "", secNum: "", level: "", primIns: "", primInsNum: "", secIns: "", secInsNum: "", assocLeagues: ""};

  /** Sets formData in initial state */
  const [formData, setFormData] = useState(INITIAL_STATE);
  
  /** Handle Submit by either creating user, updating profile, or returning an error message */

  const handleSubmit = async evt => {
    evt.preventDefault();   
    console.log("FormData in SetupProfileForm.js", formData)
    setFormData(INITIAL_STATE);
   
      // this will be if(user) then preset the form to being filled with what is already in their profile. 
     /** Update profile*/
      // const derbyName = user.derbyName;
      // const image = user.image;
      // const email = formData.email; 
      // const firstName = formData.firstName;
      // const lastName = formData.lastName;
      // const facebookName = formData.facebookName;
      // const about = formData.about;
      // const primNum = user.primNum;
      // const secNum = user.secNum; 
      // const level = user.level;
      // const primIns = user.primIns;
      // const primInsNum = user.primInsNum;
      // const secIns = user.secIns;
      // const secInsNum = user.secInsNum; 
      // const assocLeagues = user.assocLeagues;
    
      // delete formData.username;
      // delete formData.email; 
      // *Not sure what the above are doing? 

      // update(formData, derbyName);
      // let result = await FastApi.update(formData);

      // let profileData = {
      //   derbyName: derbyName,
      //   image: image, 
      //   firstName: firstName, 
      //   lastName: lastName, 
      //   email: email, 
      //   facebookName: facebookName,
      //   about: about, 
      //   primNum: primNum, 
      //   secNum: secNum, 
      //   level: level, 
      //   primIns: primIns, 
      //   primInsNum: primInsNum,
      //   secIns: secIns, 
      //   secInsNum: secInsNum, 
      //   assocLeagues: assocLeagues
      // }
      //   console.log("profileData:", profileData)
      //   // setUser(profileData)
      //   console.log("USER!!!!!:", user)
      //   setFormData(profileData);
    //     // todo: this is working but you need to relook at it as I think you made it more complicated than needed. 
    let result = await FastApi.update(formData);

    //   setValid(true)
    if(result) {
      navigate('/home')

    } else {
      // let message = result.errors[0]
      let message = result
      // setErrorMessage(message)
      // setInvalid(true)
    }
   }

  /** Update local state with current state of input element */

  const handleChange = evt => {
    console.log('handleChange is running')
    const { name, value }= evt.target;

    setFormData(fData => ({
      ...fData,
      [name]: value,
    }));
    console.log("formData:", formData)
  };

  /** toggle dropdown */

// const toggle = () => setDropdownOpen((prevState) => !prevState);

  /** render form */

  return (
    <section className="col-md-4 SetupProfileForm">
        <Card>
            <CardTitle className="SetupProfileForm-CardTitle">
            {/* { !user && ( <h1>Create a Profile</h1> )}
            { user && (<h1>{user.username}'s Profile</h1>)} */}
            <h1>Additional Information</h1>
            </CardTitle>
            <CardBody>
                {/* <Form> */}
                <Form onSubmit={handleSubmit}>
                    <FormGroup>

                    <Label htmlFor="derbyName">Derby Name: </Label>
                       <Input
                           type="text"
                           id="derbyName"
                           name="derbyName"
                           value={formData.derbyName}
                           onChange={handleChange}
                           placeholder="Derby Name"
                           required
                           // valid={valid}
                           // invalid={invalid}
                       />

                    <Label htmlFor="image">Profile Image: </Label>
                      <Input
                          type="file"
                          id="image"
                          name="image"
                          value={formData.image}
                          onChange={handleChange}
                          accept="image/png image/jpg"
                          // valid={valid}
                          // invalid={invalid}
                      />

                        <Label htmlFor="firstName">First Name: </Label>
                       
                        <Input
                            type="text"
                            id="firstName"
                            name="firstName"
                            value={formData.firstName}
                            onChange={handleChange}
                            placeholder="First Name"
                            // valid={valid}
                            // invalid={invalid}
                        />
                        
           
                        <Label htmlFor="lastName">Last Name: </Label>
                        <Input
                            type="text"
                            id="lastName"
                            name="lastName"
                            value={formData.lastName}
                            onChange={handleChange}
                            placeholder="Last Name"
                            // valid={valid}
                            // invalid={invalid}
                        />

                        <Label htmlFor="facebookName">Facebook Name: </Label>
                        <Input
                            type="text"
                            name="facebookName"
                            className="form-control"
                            value={formData.facebookName}
                            onChange={handleChange}
                            placeholder="Facebook Name"
                            id="facebookName"
  
                            // invalid={invalid}

                        />
                        {/* </> */}
                        {/* )} */}

                        <Label htmlFor="about">About: </Label>
                        <Input
                            type="textarea"
                            name="about"
                            className="form-control"
                            value={formData.about}
                            onChange={handleChange}
                            placeholder="Write your story..."
                            id="about"
  
                            // invalid={invalid}
                        />

                        <Label htmlFor="primNum">Primary Number: </Label>
                        <Input
                            type="number"
                            name="primNum"
                            className="form-control"
                            value={formData.primNum}
                            onChange={handleChange}
                            placeholder="Primary Number"
                            id="primaryNumber"
  
                            // invalid={invalid}
                        />

                        <Label htmlFor="secNum">Secondary Number: </Label>
                        <Input
                            type="number"
                            name="secNum"
                            className="form-control"
                            value={formData.secNum}
                            onChange={handleChange}
                            placeholder="Secondary Number"
                            id="secondaryNumber"
  
                            // invalid={invalid}
                            // note will have to restrict this to numbers only
                        />

                        {/* <Label htmlFor="secInsNum">Secondary Insurance Number: </Label>
                        <Input
                            type="text"
                            name="secInsNum"
                            className="form-control"
                            value={formData.secInsNum}
                            onChange={handleChange}
                            placeholder="Secondary Insurance Number"
                            id="secInsNum"
  
                            // invalid={invalid}
                            // note will have to restrict this to numbers only
                        /> */}

      
                        <Label htmlFor="level">Skill Level: </Label>
                        <Input
                            type="select"
                            name="level"
                            className="form-control"
                            value={formData.level}
                            onChange={handleChange}
                            placeholder="level"
                            id="level"
  
                            // invalid={invalid}
                            >
                              <option>
                                C
                              </option>
                              <option>
                                B
                              </option>
                              <option>
                                A
                              </option>
                              <option>
                                AA
                              </option>
                            {/* </Col> */}
                            </Input>

                        <Label htmlFor="primIns">Primary Insurance: </Label>
                        <Input
                            type="select"
                            name="primIns"
                            className="form-control"
                            value={formData.primIns}
                            onChange={handleChange}
                            placeholder="Primary Insurance"
                            id="primaryInsurance"
  
                            // invalid={invalid}
                            >
                              <option>
                                WFTDA
                              </option>
                              <option>
                                USARS
                              </option>
                              <option>
                                Other
                              </option>
              
                        </Input>

                        <Label htmlFor="primInsNum">Primary Insurance Number: </Label>
                        <Input
                            type="text"
                            name="primInsNum"
                            className="form-control"
                            value={formData.primInsNum}
                            onChange={handleChange}
                            placeholder="Primary Insurance Number"
                            id="primInsNum"
  
                            // invalid={invalid}
                            // note will have to restrict this to numbers only
                        />

                        <Label htmlFor="secIns">Secondary Insurance: </Label>
                        <Input
                            type="select"
                            name="secIns"
                            className="form-control"
                            value={formData.secIns}
                            onChange={handleChange}
                            placeholder="Secondary Insurance"
                            id="secondaryInsurance"
  
                            // invalid={invalid}
                            >
                              <option>
                                WFTDA
                              </option>
                              <option>
                                USARS
                              </option>
                              <option>
                                Other
                              </option>
                        </Input>

                        <Label htmlFor="secInsNum">Secondary Insurance Number: </Label>
                        <Input
                            type="text"
                            name="secInsNum"
                            className="form-control"
                            value={formData.secInsNum}
                            onChange={handleChange}
                            placeholder="Secondary Insurance Number"
                            id="secInsNum"
  
                            // invalid={invalid}
                            // note will have to restrict this to numbers only
                        />

                        <Label htmlFor="assocLeagues">Associated Leagues: </Label>
                        <Input
                            type="text"
                            name="assocLeagues"
                            className="form-control"
                            value={formData.assocLeagues}
                            onChange={handleChange}
                            placeholder="Associated Leagues"
                            id="assocLeagues"
  
                            // invalid={invalid}
                            // note will have to restrict this to numbers only
                        />

                        {/* todod need to add this in */}
                        {/* Played Rulesets
                        <Label htmlFor="playedRulesets">Played Rulesets: </Label>
                        <Input
                            type="radio"
                            name="WFTDA"
                            className="form-control"
                            // value={formData.primaryNumber}
                            // onChange={handleChange}
                            id="WFTDA"
  
                            // invalid={invalid}
                            // note will have to restrict this to numbers only
                        /> */}
                      
                            
                        {/* <FormFeedback valid>Profile updated successfully!</FormFeedback>
                        <FormFeedback tooltip>{errorMessage} </FormFeedback> */}

                    </FormGroup>

                    {/* { user && <Button >Save Changes</Button> }
                    { !user && <Button >Create Profile</Button> } */}
                    <Button >Save Profile</Button>
                </Form>
            </CardBody>
        </Card>
    </section>
  
);
};

export default SetupProfileForm;
