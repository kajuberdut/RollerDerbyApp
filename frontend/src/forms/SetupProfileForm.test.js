import { render, screen, cleanup } from '@testing-library/react';
import SetupProfileForm from './SetupProfileForm.js';
import { BrowserRouter } from "react-router-dom";
import '../setupTests.js';

// * Passing as of 2/13/24
// npm test BoutForm.test.js
// must be in frontend directory 

afterEach(() => {
    cleanup()
  })
  
describe("checks rendering of profile form", () => {  

    test('renders profile form page', () => {

    render(
        <BrowserRouter>
                <SetupProfileForm/>
        </BrowserRouter>);
    });


    test('matches snapshot', function() {
    const { asFragment } = render(
        <BrowserRouter>
                <SetupProfileForm />
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot(); 
    })

});

describe("asserts bout form text", () => {  

    test('renders bout form page', () => {

    render(
        <BrowserRouter>
                <SetupProfileForm />
        </BrowserRouter>
    )

        const text = screen.getByText('Setup Public Profile');
        expect(text).toBeInTheDocument();
    });

});