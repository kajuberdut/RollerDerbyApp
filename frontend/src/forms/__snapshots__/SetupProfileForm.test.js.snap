// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of profile form matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4 SetupProfileForm"
    style="margin-top: 150px;"
  >
    <div
      class="card"
      style="min-width: 400px;"
    >
      <div
        class="SetupProfileForm-CardTitle card-title"
      >
        <h1>
          Setup Public Profile
        </h1>
      </div>
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="form-label"
              for="username"
            >
              Derby Name: 
            </label>
            <input
              class="form-control"
              id="username"
              maxlength="35"
              name="username"
              placeholder="Derby Name"
              required=""
              type="text"
              value="testUser"
            />
            <label
              class="form-label"
              for="image"
            >
              Profile Image: 
            </label>
            <input
              accept="image/png image/jpg"
              class="form-control"
              id="image"
              name="image"
              type="file"
            />
            <label
              class="form-label"
              for="city"
            >
              City: 
            </label>
            <input
              class="form-control"
              id="city"
              maxlength="28"
              name="city"
              placeholder="City"
              type="text"
              value=""
            />
            <label
              class="form-label"
              for="state"
            >
              State: 
            </label>
            <select
              class="form-select"
              id="state"
              maxlength="2"
              name="state"
              placeholder="State"
            >
              <option
                value=""
              >
                State
              </option>
              <option
                value="AL"
              >
                AL
              </option>
              <option
                value="AK"
              >
                AK
              </option>
              <option
                value="AZ"
              >
                AZ
              </option>
              <option
                value="AR"
              >
                AR
              </option>
              <option
                value="CA"
              >
                CA
              </option>
              <option
                value="CO"
              >
                CO
              </option>
              <option
                value="CT"
              >
                CT
              </option>
              <option
                value="DE"
              >
                DE
              </option>
              <option
                value="DC"
              >
                DC
              </option>
              <option
                value="FL"
              >
                FL
              </option>
              <option
                value="GA"
              >
                GA
              </option>
              <option
                value="HI"
              >
                HI
              </option>
              <option
                value="ID"
              >
                ID
              </option>
              <option
                value="IL"
              >
                IL
              </option>
              <option
                value="IN"
              >
                IN
              </option>
              <option
                value="IA"
              >
                IA
              </option>
              <option
                value="KS"
              >
                KS
              </option>
              <option
                value="KY"
              >
                KY
              </option>
              <option
                value="LA"
              >
                LA
              </option>
              <option
                value="ME"
              >
                ME
              </option>
              <option
                value="MD"
              >
                MD
              </option>
              <option
                value="MA"
              >
                MA
              </option>
              <option
                value="MI"
              >
                MI
              </option>
              <option
                value="MN"
              >
                MN
              </option>
              <option
                value="MS"
              >
                MS
              </option>
              <option
                value="MO"
              >
                MO
              </option>
              <option
                value="MT"
              >
                MT
              </option>
              <option
                value="NE"
              >
                NE
              </option>
              <option
                value="NV"
              >
                NV
              </option>
              <option
                value="NH"
              >
                NH
              </option>
              <option
                value="NJ"
              >
                NJ
              </option>
              <option
                value="NM"
              >
                NM
              </option>
              <option
                value="NY"
              >
                NY
              </option>
              <option
                value="NC"
              >
                NC
              </option>
              <option
                value="ND"
              >
                ND
              </option>
              <option
                value="OH"
              >
                OH
              </option>
              <option
                value="OK"
              >
                OK
              </option>
              <option
                value="OR"
              >
                OR
              </option>
              <option
                value="PA"
              >
                PA
              </option>
              <option
                value="RI"
              >
                RI
              </option>
              <option
                value="SC"
              >
                SC
              </option>
              <option
                value="SD"
              >
                SD
              </option>
              <option
                value="TN"
              >
                TN
              </option>
              <option
                value="TX"
              >
                TX
              </option>
              <option
                value="UT"
              >
                UT
              </option>
              <option
                value="VT"
              >
                VT
              </option>
              <option
                value="VA"
              >
                VA
              </option>
              <option
                value="WA"
              >
                WA
              </option>
              <option
                value="WV"
              >
                WV
              </option>
              <option
                value="WI"
              >
                WI
              </option>
              <option
                value="WY"
              >
                WY
              </option>
            </select>
            <label
              class="form-label"
              for="about"
            >
              About: 
            </label>
            <textarea
              class="form-control form-control"
              id="about"
              maxlength="3000"
              name="about"
              placeholder="Write your story..."
            >
              This is my story...
            </textarea>
            <label
              class="form-label"
              for="primNum"
            >
              Primary Number: 
            </label>
            <input
              class="form-control form-control"
              id="primaryNumber"
              maxlength="4"
              name="primNum"
              pattern="[0-9]*"
              placeholder="Primary Number"
              type="text"
              value="01"
            />
            <label
              class="form-label"
              for="level"
            >
              Skill Level: 
            </label>
            <select
              class="form-control form-select"
              id="level"
              name="level"
              placeholder="level"
            >
              <option
                value=""
              >
                Prefer not to say.
              </option>
              <option
                value="C"
              >
                C
              </option>
              <option
                value="B"
              >
                B
              </option>
              <option
                value="A"
              >
                A
              </option>
              <option
                value="AA"
              >
                AA
              </option>
            </select>
            <label
              class="form-label"
              for="assocLeagues"
            >
              Associated Leagues: 
            </label>
            <input
              class="form-control form-control"
              id="assocLeagues"
              maxlength="200"
              name="assocLeagues"
              placeholder="Associated Leagues"
              type="text"
              value="Testing League"
            />
            <label
              class="form-label"
              for="rulesets"
            >
              Positions: 
            </label>
            <select
              class="form-control form-select"
              id="positions"
              multiple=""
              name="positions"
              placeholder="Played Positions"
            >
              <option
                value="jammer"
              >
                jammer
              </option>
              <option
                value="pivot"
              >
                pivot
              </option>
              <option
                value="blocker"
              >
                blocker
              </option>
            </select>
            <p>
              <b>
                Selected positions: 
              </b>
            </p>
            <label
              class="form-label"
              for="rulesets"
            >
              Played Rulesets: 
            </label>
            <select
              class="form-control form-select"
              id="rulesets"
              multiple=""
              name="rulesets"
              placeholder="Played Rulesets"
            >
              <option
                value="WFTDA"
              >
                WFTDA
              </option>
              <option
                value="USARS"
              >
                USARS
              </option>
              <option
                value="banked track"
              >
                Banked Track
              </option>
              <option
                value="short track"
              >
                Short Track
              </option>
            </select>
            <p>
              <b>
                Selected rulesets: 
              </b>
            </p>
            <label
              class="form-label"
              for="facebookName"
            >
              Facebook Name: 
            </label>
            <input
              class="form-control form-control"
              id="facebookName"
              maxlength="50"
              name="facebookName"
              placeholder="Facebook Name"
              type="text"
              value="facebook test user"
            />
          </div>
          <button
            class="btn btn-secondary"
          >
            Save Profile
          </button>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
